lib/tar.ml

```diff
@@ -577,6 +578,11 @@ module Header = struct
   (** Marshal a header block, computing and inserting the checksum *)
   let marshal ?level c (x: t) =
     let level = compatibility level in
+    let* () =
+      if Bytes.length c < length then
+        Error (`Msg "buffer too short")
+      else Ok ()
+    in
```

Comment:
Here I'm more inclined to raise `Invalid_argument _`. I decided to allow for longer buffers in case the caller wants to put the file contents in the same buffer. Now, maybe `marshal` could take an `?off` argument...